### PR TITLE
Handled error after updating the statefulset for ScaleMongoDBWhileBackupAndRestore  testcase: PA-1159

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
Error was not handled after updating the statefulset
**Which issue(s) this PR fixes** (optional)
Closes #https://portworx.atlassian.net/browse/PA-1159

**Special notes for your reviewer**:

### CODE REVIEW

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: tests/backup/backup_resiliency_test.go [hunk 1]
```diff
@@ -907,17 +907,17 @@ var _ = Describe("{ScaleMongoDBWhileBackupAndRestore}", func() {
 			log.InfoD("Scaling MongoDB statefulset replica to 0 while backup is in progress")
 			*statefulSet.Spec.Replicas = scaledDownReplica
 			statefulSet, err = apps.Instance().UpdateStatefulSet(statefulSet)
+			log.FailOnError(err, fmt.Sprintf("Scaling down MongoDB statefulset replica to 0"))
```
Comment (kshithijiyer-px) [inline tests/backup/backup_resiliency_test.go]: Can we use dash.verfiryfatel?

**Review by kshithijiyer-px — CHANGES_REQUESTED** [review]

File: tests/backup/backup_resiliency_test.go [hunk 2]
```diff
@@ -874,6 +874,7 @@ var _ = Describe("{ScaleMongoDBWhileBackupAndRestore}", func() {
 			dash.VerifyFatal(err, nil, "Getting backup namespace")
 			log.InfoD("Getting the replica factor of mongodb statefulset in backup namespace [%s] before taking backup", pxBackupNS)
 			statefulSet, err = apps.Instance().GetStatefulSet(mongodbStatefulset, pxBackupNS)
```
Comment (kshithijiyer-px) [inline tests/backup/backup_resiliency_test.go:876]: Can we split this into multiple lines?

File: tests/backup/backup_resiliency_test.go (hunk 1 above)
Comment (snigdha-px) [reply tests/backup/backup_resiliency_test.go]: done

File: tests/backup/backup_resiliency_test.go (hunk 2 above)
Comment (snigdha-px) [reply tests/backup/backup_resiliency_test.go:876]: As discussed, handling the error

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Comment by snigdha-px** [comment]
> Can you please add a link to a successful run?

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/1443/consoleFull

**Pull request merged**